HousingRepairsOnlineApi.Tests/HelpersTests/ServiceCollectionExtensionsTests.cs

```diff
@@ -62,35 +78,267 @@ public void GivenValidSorConfigPathArgument_WhenAddingSoREngineToServices_ThenSo
+                sorConfigurations);
+
+            // Assert
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidSorConfigurationForCreatingSorMapping))]
+        public void GivenSorConfigurations_WhenGeneratingSorMapping_ThenCorrectMappingCreated(
+            IEnumerable<SorConfiguration> sorConfigurations, IDictionary<string, IDictionary<string, dynamic>> expected)
+        {
+            // Arrange
+
+            // Act
+            var actual = ServiceCollectionExtensions.GenerateSorMapping(sorConfigurations);
+
+            // Assert
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        public static TheoryData<JsonException, string> InvalidJsonSorConfigTestData()
         {
-            yield return new object[] { new JsonSerializationException(), @"{" };
-            yield return new object[] { new JsonReaderException(), @"}" };
-            yield return new object[] { new JsonSerializationException(), @"
+            return new TheoryData<JsonException, string>
+            {
+                { new JsonSerializationException(), @"{" },
+                { new JsonReaderException(), @"}" },
+                {
+                    new JsonSerializationException(), @"{
+                        ""kitchen"": {
+                    }"
+                },
                 {
-                    ""kitchen"": {
-                }"
+                    new JsonReaderException(), @"[{
+                    ""value"": ""kitchen"",
+                    ""display"": ""Kitchen"",
+                    ""options"": [{
+                            ""value"": ""cupboards"",
+                            ""display"": ""Cupboards, including damaged..."",
+                            ""options"": [{
+                                ""value"": ""doorHangingOff"",
+                                ""display"": ""Hanging Door"",
+                                ""sorCode"": ""SoR Code 1""
+                            }]
+                        }]
+                    {
+                    ""value"": ""bathroom"",
+                    ""display"": ""Bathroom"",
+                    ""options"": [{
+                        ""value"": ""bath"",
+                        ""display"": ""Bath, including taps"",
+                        ""options"": [{
+                            ""value"": ""taps"",
+                            ""display"": ""Bath taps"",
+                            ""sorCode"": ""SoR Code 1""
+                        }]
+                    }]
+                }]"
+                },
             };
-            yield return new object[] { new JsonReaderException(), @"
+        }
+
+        public static TheoryData<IEnumerable<SorConfiguration>, IEnumerable<RepairTriageOption>> ValidSorConfigurationForCreatingJourneyRepairTriageOptions()
+        {
+            var value = "value1";
```

Comment:
I've got prompts here from my code editor to change these into `constant`s.

Reply:
Extracted to field constants